Fix minor typos and clarify non-blocking handler guidance (#975)
eclipse-lsp4j/lsp4j: documentation/README.md

> When implementing the handlers for requests or notifications, you need to be aware that the calling thread is the thread that reads and dispatches incoming messages. Therefore, blocking it may result in reduced throughput or even a deadlock (https://github.com/eclipse-lsp4j/lsp4j/issues/775). As a general rule, message handlers should be implemented in a non-blocking, asynchronous way.
> 
> For long-running work, prefer returning a `CompletableFuture` and performing the actual computation outside of the message dispatching thread. This keeps the JSON-RPC message processor responsive while the request is being handled asynchronously.

Reviewer: I think that using a code example could be the best way to clarify it, e.g. based on the actual case from #775:

````suggestion
When implementing the handlers for requests or notifications, you need to be aware that the calling thread is the thread that reads and dispatches incoming messages. Therefore, blocking it may result in reduced throughput or even a deadlock (https://github.com/eclipse-lsp4j/lsp4j/issues/775). As a general rule, message handlers should be implemented in a non-blocking, asynchronous way. For example, instead of

```java
@Override
public void didChangeWorkspaceFolders(DidChangeWorkspaceFoldersParams params) {
    build();
}
```

use

```java
@Override
public void didChangeWorkspaceFolders(DidChangeWorkspaceFoldersParams params) {
    CompletableFuture.runAsync(() -> {
        build();
    });
}
```
````

A separate example along these lines could also be added for a request handler, if necessary.
